mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2220,48 +2219,67 @@ impl ElectrumClient {
 #[cfg_attr(test, mockable)]
 impl UtxoRpcClientOps for ElectrumClient {
     fn list_unspent(&self, address: &Address, _decimals: u8) -> UtxoRpcFut<Vec<UnspentInfo>> {
-        let script = try_f!(output_script(address));
-        let script_hash = electrum_script_hash(&script);
-        Box::new(
-            self.scripthash_list_unspent(&hex::encode(script_hash))
-                .map_to_mm_fut(UtxoRpcError::from)
-                .map(move |unspents| {
+        let mut output_scripts = vec![try_f!(output_script(address))];
+
+        // If the plain pubkey is available, fetch the UTXOs found in P2PK outputs as well (if any).
+        if let Some(pubkey) = address.pubkey() {
```

Comment:
should we not add the same code into fn list_unspent_group() too?

Reply:
@mariocynicys any comments about adding p2pk to fn display_balances and list_unspent_group?

Reply:
Yeah, since we settled on HD wallets not having legacy p2pk outputs (we won't support this case), this poll isn't needed for the `_group` methods. That's to avoid increasing the traffic on the electrum servers.